src/inmanta/execute/runtime.py

```diff
@@ -173,18 +173,29 @@ def get_dataflow_node(self) -> dataflow.AssignableNodeReference:
         return self._node
 
 
-class AttributeVariable(ResultVariable["Instance"]):
+class RelationAttributeVariable(ABC):
```

Comment:
Doesn't extending ABC have a significant performance overhead?  (For no reason that could be achieved by documentation alone),?


(these objects are constructed by the thousands)

Reply:
I don't immediately see how, but you're right that there's no strong reason to do so, I'll get rid of it.